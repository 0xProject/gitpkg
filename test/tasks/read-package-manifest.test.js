import path from 'path';
import readPackageManifest from '../../src/tasks/Task/read-package-manifest';

const invalidsPath = path.resolve(__dirname, '../fixtures/package-json-files');

describe('while using readPackageManifest()', () => {
  it('should throw the proper error with package.json with invalid version', async () => {
    try {
      await readPackageManifest(path.resolve(invalidsPath, 'invalid-version'));
    } catch (e) {
      expect(() => {
        throw e;
      }).toThrowErrorMatchingSnapshot();
    }
  });
  it('should throw the proper error with package.json with no name', async () => {
    try {
      await readPackageManifest(path.resolve(invalidsPath, 'no-name'));
    } catch (e) {
      expect(() => {
        throw e;
      }).toThrowErrorMatchingSnapshot();
    }
  });
  it('should throw the proper error with package.json with no version', async () => {
    try {
      await readPackageManifest(path.resolve(invalidsPath, 'no-version'));
    } catch (e) {
      expect(() => {
        throw e;
      }).toThrowErrorMatchingSnapshot();
    }
  });
  it('should read a valid package.json', async () => {
    const validPkg = await readPackageManifest(path.resolve(invalidsPath, 'valid'));
    expect(typeof validPkg).toBe(typeof {});
    expect(typeof validPkg.name).toBe(typeof '');
    expect(typeof validPkg.version).toBe(typeof '');
  });
});
