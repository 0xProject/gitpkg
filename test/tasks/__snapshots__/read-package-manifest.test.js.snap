// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`while using readPackageManifest() should throw the proper error with package.json with invalid version 1`] = `"Invalid semver version."`;

exports[`while using readPackageManifest() should throw the proper error with package.json with no name 1`] = `"Package doesn't have a name."`;

exports[`while using readPackageManifest() should throw the proper error with package.json with no version 1`] = `"Package doesn't have a version."`;
